common/src/main/java/org/apache/celeborn/common/network/server/TransportServer.java

```diff
@@ -44,17 +47,25 @@ public class TransportServer implements Closeable {
 
   private final TransportContext context;
   private final TransportConf conf;
-
+  private final BaseMessageHandler appMessageHandler;
+  private final List<TransportServerBootstrap> bootstraps;
   private ServerBootstrap bootstrap;
   private ChannelFuture channelFuture;
   private AbstractSource source;
   private int port = -1;
 
   public TransportServer(
-      TransportContext context, String hostToBind, int portToBind, AbstractSource source) {
+      TransportContext context,
+      String hostToBind,
+      int portToBind,
+      AbstractSource source,
+      BaseMessageHandler appMessageHandler,
+      List<TransportServerBootstrap> bootstraps) {
     this.context = context;
     this.conf = context.getConf();
     this.source = source;
+    this.appMessageHandler = Preconditions.checkNotNull(appMessageHandler);
+    this.bootstraps = Lists.newArrayList(Preconditions.checkNotNull(bootstraps));
```

Comment:
```suggestion
    this.bootstraps = Preconditions.checkNotNull(bootstraps);
```

Reply:
Creating a new list seems better to me. The passed in `bootstraps` reference will be available outside `TransportServer` and can be modified. This ensures that the `this.bootstraps` will not change after the `TransportServer` is initialized.